### ISSUE TITLE
Dns.GetHostEntryAsync is faulty on Windows

### ISSUE THREAD

**Issue by jameyboor** [dotnet/corefx#22589]
When calling `Dns.GetHostEntryAsync()` an  `AggregateException` is always thrown with information of:

> System.AggregateException was unhandled
>   Source=System.Private.CoreLib
>   InnerException: 
>        HResult=-2147467259
>        Message=Either the application has not called WSAStartup, or WSAStartup failed
>        NativeErrorCode=10093
>        Source=System.Net.NameResolution
>        StackTrace:
>             at System.Net.Dns.HostResolutionEndHelper(IAsyncResult asyncResult)
>             at System.Net.Dns.EndGetHostEntry(IAsyncResult asyncResult)
>             at System.Net.Dns.<>c.<GetHostEntryAsync>b__15_1(IAsyncResult asyncResult)
>             at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
> 

It seems that the method does not correctly initialize Winsock through `WSAStartup()`.

A hacky "fix" at this moment is asking for a response from a WebRequest first to ensure that Winsock is initialized:
```
var req = WebRequest.Create("http://xxxxxx.com");
var res = req.GetResponseAsync().Result;
IPHostEntry hostInfo = Dns.GetHostEntryAsync(hostIPAddress).Result;
```

**Comment by stephentoub** [comment]
What version of .NET Core are you using? I seem to remember that being a bug in 1.0 that was subsequently fixed.

**Comment by stephentoub** [comment]
This was fixed here for 1.1:
https://github.com/stephentoub/corefx/commit/2a7dcbbaf79cdde8146622ee9fe14947e7211e06
